.github/workflows/build.yml

```diff
@@ -192,10 +201,36 @@ jobs:
         run: |
           apk add --no-cache alpine-sdk bash clang curl gn lld llvm python3
 
+      - name: Switch to alpine edge
+        # note that we're switching after installing packages for host,
+        # but before handling sysroot. this is needed because riscv64 is edge only
+        if: matrix.target-arch == 'riscv64'
+        run: |
+          echo "
+          https://dl-cdn.alpinelinux.org/alpine/edge/main
+          https://dl-cdn.alpinelinux.org/alpine/edge/community
+          " > /etc/apk/repositories
+
       - name: Install sysroot
         run: |
           cd dart-sdk/sdk
-          ./build/linux/alpine_sysroot_scripts/install-sysroot.sh
+          dart_to_alpine_arch() {
```

Comment:
While it is nice trying to install only what's absolutely needed, installing extra sysroots only takes a few seconds. Comparing to that the whole job takes about 2-3 hours on GitHub Action, it not worth saving a few seconds.

Therefore, I would prefer just keep installing all variants of sysroot for simplicity. This is also what the official dart sdk does for its debian sysroots that all of them get installed regardless of what is going to be compiled.

As for `riscv64`, you can just add an extra step to install it for now. Once `riscv64` lands in alpine stable, we can send the patch to upstream.

Reply:
ok, PR simplified then.

the patch to dart (https://dart-review.googlesource.com/c/sdk/+/296901) makes the script accept a list of architectures or by default download the previous list of architectures (without riscv64) so I'm planning to submit that as it is - I don't think riscv64 is getting shipped anytime soon in alpine, 3.18 currently being built definitely will not

Reply:
The patch looks good.